src/stirling/utils/linux_headers.cc

```diff
@@ -498,30 +498,26 @@ Status LinkHostLinuxHeaders(const std::filesystem::path& lib_modules_dir) {
   return Status::OK();
 }
 
-Status ExtractPackagedHeaders(PackagedLinuxHeadersSpec* headers_package) {
-  std::string expected_directory =
-      absl::Substitute("/usr/src/linux-headers-$0.$1.$2-pl", headers_package->version.version,
-                       headers_package->version.major_rev, headers_package->version.minor_rev);
-
+Status ExtractPackagedHeaders(PackagedLinuxHeadersSpec* headers_package,
+                              std::string staging_directory, std::string expected_directory) {
   // This is a loose check that we don't clobber what we *think* should the output directory.
   // If someone built a tar.gz with an incorrect directory structure, this check wouldn't save us.
   if (fs::Exists(expected_directory)) {
     return error::Internal("Refusing to clobber existing directory: $0.", expected_directory);
   }
-
-  // Extract the files.
+  std::filesystem::create_directories(staging_directory);
+  // Extract the files from the tarball, stripping the leading prefix
+  // "usr/src/linux-headers-$0.$1.$2-pl" to avoid unnecessary nesting in the staging directory.
   ::px::tools::Minitar minitar(headers_package->path.string());
-  PX_RETURN_IF_ERROR(minitar.Extract("/"));
-
-  // Check that the expected path was created.
-  if (!fs::Exists(expected_directory)) {
+  PX_RETURN_IF_ERROR(minitar.Extract(staging_directory, expected_directory.substr(1)));
```

Comment:
For clarity, would suggest a local like this:
`const std::string prefix_to_strip = expected_directory.substr(1);`